mlflow/projects/_project_spec.py

```diff
@@ -83,6 +83,18 @@ def _validate_parameters(self, user_parameters):
                 "No value given for missing parameters: %s" %
                 ", ".join(["'%s'" % name for name in missing_params]))
 
+    def consolidate_parameters(self, user_parameters):
```

Comment:
How is this different from compute_parameters? I thought that was supposed to get the final ones given user parameters too. It's pretty confusing to have these two methods with fairly similar names.

Reply:
`compute_parameters` is used during the execution phase and other than consolidation of default and user specified params on CLI, it also does a local download for `path` type params and does a substitution. For logging params we would need original paths.

Also, `compute_...` needs to split up params as `MLproject` listed params (for command line substitution) and additional params to be sent in as ``--param_key value``.

They are kinda different. Lemme see if there is a cleaner way to refactor this to avoid confusion.